### PR TITLE
Update fetch file to centralize default headers

### PR DESCRIPTION
**Pull request by yashkohli88**
### Description:
This pull request focuses on refactoring the code to centralize the declaration of default headers in a single `fetch.js` file, updating all fetch functions in various package managers across the project to use the centralized user-agent header, and adding tests to validate these changes. Below are the tasks accomplished:

### Checklist:
- [x] Refactored code to centralize default headers in `fetch.js`.
- [x] Updated all fetch functions to use the centralized default headers.
- [x] Added tests to validate the changes.
- [x] Ran all existing and new tests to ensure correctness.

This PR is related to issue - https://github.com/clearlydefined/crawler/issues/619

### CODE REVIEW

File: test/unit/lib/fetchTests.js [hunk 1]
```diff
@@ -1,9 +1,14 @@
 const { fail } = require('assert')
-const { callFetch, withDefaults } = require('../../../lib/fetch')
+const { callFetch, withDefaults, defaultHeaders } = require('../../../lib/fetch')
 const { expect } = require('chai')
 const fs = require('fs')
 const mockttp = require('mockttp')
 
+function checkHeader(headers) {
```
Comment (qtomlinson) [inline test/unit/lib/fetchTests.js:7]: checkDefaultHeader?

File: test/unit/providers/fetch/mavenBasedFetchTests.js [hunk 2]
```diff
@@ -2,8 +2,15 @@ const { expect } = require('chai')
 const MavenBasedFetch = require('../../../../providers/fetch/mavenBasedFetch')
 const mockttp = require('mockttp')
 const sinon = require('sinon')
+const { defaultHeaders } = require('../../../../lib/fetch')
 const Request = require('../../../../ghcrawler').request
 
+function checkHeader(headers) {
```
Comment (qtomlinson) [inline test/unit/providers/fetch/mavenBasedFetchTests.js:8]: checkDefaultHeader for clarity?

**Review by qtomlinson — COMMENTED** [review]
Thanks for the refactoring!

**Review by ljones140 — APPROVED** [review]
Looks great.
nice improvement to make sure we have consistent headers

**Pull request merged**